build.gradle

```
@@ -71,16 +70,17 @@ task generateEnums(type: Exec, dependsOn: compileEnumGenerator) {
 /**
  * Generates the JNI class headers
  */
-task generateJniHeaders(type: Exec, dependsOn: [generateEnums, compileJava]) {
-    def classesDir = sourceSets.main.output.classesDir
-    if (!classesDir.exists()) classesDir.mkdirs()
-
-    commandLine 'javah',
-            '-d',
-            "${projectDir}/src/main/c",
-            '-classpath',
-            classesDir,
-            'pt.davidafsilva.apple.OSXKeychain'
+task generateJniHeaders(type: JavaCompile, dependsOn: [generateEnums]) {
+    classpath = sourceSets.main.compileClasspath
+    destinationDir file("${buildDir}/generated/jni")
+    source = sourceSets.main.java
+    options.compilerArgs += [
+            '-h', file("${projectDir}/src/main/c"),
+            '-d', file("${buildDir}/generated/jni-tmp")
+    ]
+    doLast {
+        delete file("${buildDir}/generated/jni-tmp")
+    }
```

I don't think you need to specify the output directory for the classes as we're only generating headers here.

```suggestion
    options.compilerArgs += [
        '-h', "${projectDir}/src/main/c"
    ]
```